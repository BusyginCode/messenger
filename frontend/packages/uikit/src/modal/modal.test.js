import React from 'react'
import { shallow } from 'enzyme'
import Modal from './modal'

describe('uikit/Modal', () => {
  it('should render correctly', () => {
    expect(
      shallow(
        <Modal name="test" isOpened leftHeaderControl="div" rightHeaderControl="div">
          Modal
        </Modal>
      )
    ).toMatchSnapshot()
  })

  it('should render correctly closed', () => {
    expect(shallow(<Modal>Modal</Modal>)).toMatchSnapshot()
  })

  it('should render correctly isCustom', () => {
    expect(
      shallow(
        <Modal name="test" isOpened isCustom>
          Modal
        </Modal>
      )
    ).toMatchSnapshot()
  })
})
