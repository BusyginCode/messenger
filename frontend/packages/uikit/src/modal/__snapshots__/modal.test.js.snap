// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Modal should render correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="general"
  closeTimeoutMS={0}
  contentLabel=""
  isOpen={true}
  overlayClassName="overlay"
  overlayRef={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal"
    data-role="modal-test"
  >
    <div
      className="content"
      data-role="modal-content-test"
    >
      <Component
        getOverlayRef={[Function]}
        renderHeader={[Function]}
      />
    </div>
  </div>
  <Svg(close)
    className="close"
    data-role="modal-close-icon"
  />
</Modal>
`;

exports[`uikit/Modal should render correctly closed 1`] = `""`;

exports[`uikit/Modal should render correctly isCustom 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="custom"
  closeTimeoutMS={0}
  contentLabel=""
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  Modal
</Modal>
`;
